src/commands/ml-models-delete.ts

```diff
@@ -1,9 +1,33 @@
+import * as clc from "cli-color";
 import { Command } from "../command";
+import * as getProjectId from "../getProjectId";
+import { requirePermissions } from "../requirePermissions";
+import { ensureFirebaseMlApiEnabled, isValidModelId } from "../ml/mlHelper";
+import * as mlApi from "../ml/mlApi";
+import * as utils from "../utils";
 
 export default new Command("ml:models:delete <modelId>")
-  .description("deletes the Firebase ML Model with the given modelId")
+  .description("delete the Firebase ML Model with the given modelId")
+  .help("TODO")
+  .before(requirePermissions, ["firebaseml.models.delete"])
+  .before(ensureFirebaseMlApiEnabled)
   .action(
-    async (modelId: string): Promise<void> => {
-      throw new Error("Not Implemented");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async (modelId: string, options: any): Promise<void> => {
+      if (!isValidModelId(modelId)) {
+        return utils.reject("Must specify a valid model ID", { exit: 1 });
```

Comment:
prefer throwing a FirebaseError rather than using `utils.reject` - they do something similar, but throwing is more clear (IMO). The signature looks pretty much the same; `FirebaseError` lives in `src/error.ts`

```suggestion
        throw new FirebaseError("Must specify a valid model ID");
```

`exit: 1` is the default option for FirebaseError too, so that can be dropped!

(Same below)

(I should put this in the docs somewhere)

Reply:
Nice. Thanks